watertap/unit_models/tests/test_crystallizer_surrogate.py

```diff
@@ -0,0 +1,236 @@
+
+    # Feed composition
+    feed = {
+        "ion_composition_g_kg": {
+            "Na_+": 91.49859876,
+            "K_+": 3.185931759,
+            "Cl_-": 159.0941278,
+            "Ca_2+": 0.788476605,
+            "Mg_2+": 10.5806906,
+            "HCO3_-": 1.614773395,
+            "SO4_2-": 22.20893266,
+        },
+        "water_content_kg": 12.0764972138554,
+    }
+
+    g_to_kg = 1e-3
+    for i in m.fs.cryst_prop_feed.component_list:
+        if i == "H2O":
+            m.fs.cryst.inlet.flow_mass_phase_comp[0.0, "Liq", i].fix(
+                feed["water_content_kg"]
+            )
+        else:
+            m.fs.cryst.inlet.flow_mass_phase_comp[0.0, "Liq", i].fix(
+                feed["water_content_kg"] * feed["ion_composition_g_kg"][i] * g_to_kg
+            )
+
+    m.fs.cryst.inlet.temperature.fix(298.15)
+    m.fs.cryst.inlet.pressure.fix(101325)
+    print(
+        "Degrees of freedom of crystallizer model with defined feed",
+        degrees_of_freedom(m),
+    )
+    assert_units_consistent(m)
+
+    # Create dummy variables for inputs and outputs
+    surrogate_inputs = {
+        "Temperature": {
+            "flowsheet_var": m.fs.cryst.temperature_operating,
+            "var_bounds": (303.15, 372.15),
+        },
+        "Evaporation percent": {
+            "flowsheet_var": m.fs.cryst.evaporation_percent,
+            "var_bounds": (30.0, 98.0),
+        },
+    }
+    m.fs.o1 = Var(units=pyunits.atm)
+    m.fs.o2 = Var(units=pyunits.g / pyunits.s)
+    m.fs.o3 = Var(units=pyunits.g / pyunits.s)
+    m.fs.o4 = Var(units=pyunits.g / pyunits.s)
+    m.fs.o5 = Var(units=pyunits.g / pyunits.s)
+    surrogate_outputs = {
+        "Vapor Pressure (atm)": {"flowsheet_var": m.fs.o1, "Solid": False},
+        "Calcite_g": {"flowsheet_var": m.fs.o2, "Solid": True},
+        "Anhydrite_g": {"flowsheet_var": m.fs.o3, "Solid": True},
+        "Glauberite_g": {"flowsheet_var": m.fs.o4, "Solid": True},
+        "Halite_g": {"flowsheet_var": m.fs.o5, "Solid": True},
+    }
+
+    # if surrogate_model_form == "RBF":
+
+    add_crystallizer_rbf_model(
+        m.fs.cryst,
+        surrogate_inputs_with_bounds=surrogate_inputs,
+        surrogate_outputs=surrogate_outputs,
+        surrogate_to_flowsheet_basis_ratio=1,
+    )
+    # elif surrogate_model_form == "NN":
+    #     print("\nNN models selected.\n")
+    #     add_crystallizer_nn_model(
+    #         m.fs.cryst,
+    #         surrogate_inputs_with_bounds=surrogate_inputs,
+    #         surrogate_outputs=surrogate_outputs,
+    #         path_to_onnx_model=r"C:\Users\OOAmusat-II\Desktop\NAWI\WATERTAP\Phreeqc\crystallization\NN_40_NL_3_sigmoid_WD_1e-06_LR_0.01_testing.onnx",
+    #         path_to_offsetscaler_json=r"C:\Users\OOAmusat-II\Desktop\NAWI\WATERTAP\Phreeqc\crystallization\scalingfile_testing.json",
+    #         surrogate_to_flowsheet_basis_ratio=1,
+    #     )
+    # else:
+    #     raise ValueError('"surrogate_model_form" must be "RBF" or "NN"')
```

Comment:
Can probably test the NN surrogate form as well but just trying rbf first.

Reply:
I agree.

Reply:
I am going to exclude this for now since it might not be a priority given latest developments.